packages/headless/src/api/knowledge/stream-answer-api.ts

```diff
@@ -148,14 +144,37 @@ const handleEndOfStream = (
   draft.isStreaming = false;
 };
 
+interface MessageType {
+  payloadType: PayloadType;
+  payload: string;
+  finishReason?: string;
+  errorMessage?: string;
+  code?: number;
+}
+
+const handleError = (draft: GeneratedAnswerStream, message: MessageType) => {
```

Comment:
If you're 100% sure all `MessageType` properties are going to be defined on the `message`  argument when `handleError` is called (which seems like a fair assumption!) you could do:

```
const handleError = (draft: GeneratedAnswerStream, message: Required<MessageType>) => {
```

and then omit the `!` when setting `draft.error`.

Reply:
I cannot believe I never used the Required generic.